src/Command/Pull/PullCommandBase.php

```diff
@@ -256,6 +266,51 @@ protected function downloadDatabaseDump(
     return $local_filepath;
   }
 
+  /**
+   * Create an on-demand backup and wait for it to become available.
+   *
+   * @param \AcquiaCloudApi\Response\EnvironmentResponse $environment
+   * @param \stdClass $database
+   * @param $acquia_cloud_client
+   */
+  protected function createBackup(EnvironmentResponse $environment, \stdClass $database, $acquia_cloud_client): void {
+    $backups = new DatabaseBackups($acquia_cloud_client);
+    $response = $backups->create($environment->uuid, $database->name);
+    $url_parts = explode('/', $response->links->notification->href);
+    $notification_uuid = end($url_parts);
+    $this->waitForBackup($notification_uuid, $acquia_cloud_client);
+  }
+
+  /**
+   * Wait for an on-demand backup to become available (Cloud API notification).
+   *
+   * @param string $notification_uuid
+   * @param $acquia_cloud_client
+   */
+  protected function waitForBackup($notification_uuid, $acquia_cloud_client): void {
+    $loop = Factory::create();
+    $spinner = LoopHelper::addSpinnerToLoop($loop, 'Waiting for database backup to complete...', $this->output);
+    $notifications = new Notifications($acquia_cloud_client);
+
+    $loop->addPeriodicTimer(5, function () use ($loop, $spinner, $notification_uuid, $notifications) {
+      try {
+        $response = $notifications->get($notification_uuid);
+        if ($response->status === 'completed') {
+          LoopHelper::finishSpinner($spinner);
+          $loop->stop();
+          $this->output->writeln('');
+          $this->output->writeln('<info>Database backup is ready!</info>');
```

Comment:
Seems like we might want to print the UUID of the backup or a link to it, if that's available.

Reply:
It's not available (ping me in Slack and I can send you the internal ticket for this)